sds_data_manager/sds_cognito_stack.py

```diff
@@ -0,0 +1,131 @@
+import os
+
+import aws_cdk as cdk
+import aws_cdk.aws_cognito as cognito
+import aws_cdk.aws_iam as iam
+import aws_cdk.aws_lambda as lambda_
+import aws_cdk.aws_lambda_python_alpha as lambda_alpha_
+from aws_cdk import (
+    Stack,
+)
+from constructs import Construct
+
+
+class SdsCognitoStack(Stack):
+    def __init__(
+        self,
+        scope: Construct,
+        construct_id: str,
+        sds_id: str,
+        initial_email: str = "",
+        **kwargs,
+    ) -> None:
+        super().__init__(scope, construct_id, **kwargs)
+
+        ########### INIT
+        initial_user_context = initial_email
+
+        ########### IAM POLICIES
+        cognito_admin_policy = iam.PolicyStatement(
+            effect=iam.Effect.ALLOW,
+            actions=["cognito-idp:*"],
+            resources=["*"],
+        )
+
+        ########### COGNITO
+        # Create the Cognito UserPool
+        userpool = cognito.UserPool(
+            self,
+            id="TeamUserPool",
+            user_pool_name=f"sds-userpool-{sds_id}",
+            account_recovery=cognito.AccountRecovery.EMAIL_ONLY,
+            auto_verify=cognito.AutoVerifiedAttrs(email=True),
+            standard_attributes=cognito.StandardAttributes(
+                email=cognito.StandardAttribute(required=True)
+            ),
+            sign_in_aliases=cognito.SignInAliases(username=False, email=True),
+            removal_policy=cdk.RemovalPolicy.DESTROY,
+        )
+
+        # Add a client sign in for the userpool
+        command_line_client = cognito.UserPoolClient(
+            user_pool=userpool,
+            scope=self,
+            id="sds-command-line",
+            user_pool_client_name=f"sdscommandline-{sds_id}",
+            id_token_validity=cdk.Duration.minutes(60),
+            access_token_validity=cdk.Duration.minutes(60),
+            refresh_token_validity=cdk.Duration.minutes(60),
+            auth_flows=cognito.AuthFlow(
```

Comment:
I'm not following what all of this does, but anything with admin always screams out to me. Should we only be using the srp workflow?

https://docs.aws.amazon.com/cognito/latest/developerguide/amazon-cognito-user-pools-authentication-flow.html#amazon-cognito-user-pools-admin-authentication-flow